### ISSUE TITLE
Styling Guide and Best Practices

### ISSUE THREAD

**Issue by Jantoom** [UQdeco2800/2021-studio-7#54]
# Description
 
After Sprint 1, it was evident that no standard had been set for new additions to the codebase. To solve this, a guide should be written on how to keep the file system neat, and code readable/navigable by other studio members.
 
# Dependencies
- Restructuring of game files
 
# Milestones
- [x] Document new file structuring guide (Sep. 17)
- [x] Document new symbol declaration guide (Sep. 17)
- [x] Document new concepts that should be used throughout the codebase (Sep. 17)
 
**Completion Deadline:** End of Sprint 2
 
# Documentation
- [Styling Guide and Best Practices](https://github.com/UQdeco2800/2021-studio-7/wiki/Styling-Guide-and-Best-Practices-for-Studio-7)
 
# Members
You can add yourself if you contribute at all to the guide

- Jaleel Abdur-Raheem (@Jantoom) (Jaleel Abdur-Raheem)

**Comment by ChanghaoXi** [comment]
Hello, @Jantoom. This is Raymond from team 1, I have read through the awesome styling guide written by u. Just have one question regarding the naming of character images, only stand/standing ones are covered, can I use 'move/moving' for sprites that used to demonstrate a moving status? Thanks in advance!

**Comment by Jantoom** [comment]
Hey! I guess my intention with the naming convention was to separate images used for the TextureRenderComponent and the AnimationRenderComponent by making animation names "doing" words. This isn't explicitly in the wiki but I think that was the logic I was going by when creating it. I do think it is more accurate to describe character movement animations as "moving" rather than "move". To me, "moving" implies several transitions within a state, whereas "move" implies several states. I'll admit it is gratuitous depending on the individual's definition.

Regarding the capitalisating of .png files, I don't think they should be capitalised.  My main reasoning is that files in assets/ aren't Java classes, and there isn't any benefit to having them capitalised. If we aren't distinguishing Java and non-Java files, then it would make sense that every file type is capitalised. I personally prefer distinguishing them as an aesthetic choice. There are more reasons but I don't want to write them all haha

If you disagree with anything, come to a consensus with your team (and/or anyone else affected) and change the documentation as they see fit. The documentation's not perfect, but it should be finalised by this sprint.

Hope this helped :)

**Comment by ChanghaoXi** [comment]
@Jantoom Thank you very much for the clarifications, they are really really helpful! So basically, the styling guide is mainly used to organising different files and put them under different repositories for future reference, while as long as the file name is readable and understandable, it should be okay, right?

**Comment by Jantoom** [comment]
Depends on if you think file names should be used for future reference as well. 

It's a matter of whether your naming convention is the _most_ readable and understandable (something that we should absolutely strive for). The biggest thing that helps that is standardising the convention everywhere it can be used. It's strict, but it is by far the most concise way to convey meaning to other people that will be looking at your files.

For example, let's say you wanted to use {up, down, left, right} instead of {north, east, south, west}. You change your characters to that convention, but objects and tiles that are orientation-dependent still use NESW. Down the line, this can create confusion to anyone that looks at both implementations (from a file viewing **and** code perspective). If you were building upon these two conventions, you would have to actively think about which one to use. This creates an opportunity for bugs to occur, and someone will inevitably mess this up.

You could argue that where we draw the line is arbitrary, but I believe that thing's like capitalisation fall clearly in the "should standardise" zone.

In conclusion, the styling guide is used for **everything** that can be standardised. File organisation and file naming are **both** used for future reference, so they should both be under some form of standardisation. If you are dead-set on using "move" and capitalisation, make sure most people using animations and .png files are on-board and change the documentation as they see fit.

**Comment by ChanghaoXi** [comment]
@Jantoom Understood, I think I would have NESW and moving/standing with lowercase letters etc. then, thank you very much for your answer :)
